### ISSUE TITLE
Feature-Request: Enable Math in Queries using Cloudwatch Datasource

### ISSUE THREAD

**Issue by mnietz** [grafana/grafana#5390]
I try to use latest Grafana-Version with Cloudwatch Datasource to graph the Read and Write Units / sec on our Dynamodb.

I'm able to graph the sum of all Events, but to get the Units / sec, simple Math or a derivative Function would be necessary to get Units/sec

I would love to see this Feature in one of the upcoming Releases

**Comment by cmclaughlin** [comment]
This would be really nice to have!

I just added CloudWatch as a data source for Disk operations hoping to graph IOPS.  But since the CloudWatch polling interval is 1 or 5 minutes, you have to divide to calculate IOPS.  Being able to do math on CloudWatch metrics would solve the problem for me.

**Comment by mtanda** [comment]
This is required for CloudWatch alerting feature for Grafana (https://github.com/grafana/grafana/issues/5858).

For example, if we set alert for ratio of dynamodb consumed/provisioned capacity, we need some calculation.
```
(ConsumedWriteCapacityUnits_Sum / Period) / ProvisionedWriteCapacityUnits_Average > 0.8
```

This should be done in datasource itself.

**Comment by J-Mx** [comment]
+1 ! Really need it !

**Comment by torkelo** [comment]
closing this in favor general purpose grafana transformation functions (meta queries) https://github.com/grafana/grafana/issues/3677

**Comment by torkelo** [comment]
Cloudwatch does not support that feature, it’s query api is very limited

**Comment by torkelo** [comment]
And the meta queries feature does not exist yet so no documented

**Comment by marefr** [comment]
@phspagiari Math expressions is actually supported since #11487 was fixed. It will be included in Grafana v5.3.0 stable. You can try it out now using a nightly build.

**Comment by Leigan0** [comment]
We have Grafana v5.3.1 (08c7908) but I cannot see the math functions. Is this available for this version? Thanks

**Comment by marefr** [comment]
@Leigan0 Basically there are two new fields, Id and Expression. You can see examples of how to use it in the PR which implemented it (#11624).

**Comment by houssem-yahiaoui** [comment]
I confirm that this is working in v5.4.3 and for reference anyone trying to figure out this check : #11624
